src/libraries/System.IO.Compression.Brotli/src/System/IO/Compression/dec/BrotliStream.Decompress.cs

```diff
@@ -227,9 +234,23 @@ private bool TryDecompress(Span<byte> destination, out int bytesWritten)
             return false;
         }
 
+        private static bool GetStrictValidationSetting()
+        {
+            if (AppContext.TryGetSwitch("System.IO.Compression.UseStrictValidation", out bool strictValidation))
+            {
+                return strictValidation;
+            }
+
+            string? envVar = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_COMPRESSION_STRICT_VALIDATION");
+            return envVar is not null && (envVar == "1" || envVar.Equals("true", StringComparison.OrdinalIgnoreCase));
+        }
```

Comment:
This should be cached into a static readonly bool.

Reply:
I considered that but thought we wanted to be able to change that during runtime. Guess not. Changed.